.changeset/odd-tomatoes-juggle.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-github': patch
+---
+
+Added Github Entity Provider functionality for adding entities to the catalog.
```

Comment:
would be great to include more information about how to migrate from processor to provider.

Reply:
@pjungermann you mentioned in a previous comment you did some processors -> provider migrations how did they go ? Without much thinking something came to my mind is maybe with a raw SQL to delete all entries from the refresh_state table where the location key contains the github-discovery prefix ? After that if you register the new provider it should discover exactly the same entities for you

Reply:
I ended up removing the processor first, cleaned up the data, and then I added the provider which was re-registering the entities.

On discord, there was the comment by one of the maintainers that it is possible to set the locationKey to null at the DB. Then, it will not be considered a conflict and the provider can claim/replace it.

As processors can create orphans, you either take care of these before or you just go for removing all data.

Reply:
PS: with my comment I meant the code part (what to replace with what; config, catalog.ts, dependencies, etc.). But adding something about the migration process considering the data makes sense, too.